### PR TITLE
[13.0][ENH] agreement_legal

### PR DESCRIPTION
**Pull request by etobella**
This PR applies several commits found on 14 that were not present on 13.0

#669 #696 #700 #725

TODO: proably it is needed to add a migration script for the change from product.template to agreement.line

### CODE REVIEW

**Comment by etobella** [comment]
@newtratip @patrickrwilson @victoralmau

**Comment by pedrobaeza** [comment]
#725 should be already in 13.0

**Comment by etobella** [comment]
Sorry @pedrobaeza You are right, I had to remake commits in order to check that everything working fine. I think now it could be right

**Comment by pedrobaeza** [comment]
About the `is_old_version` things, it's worse solution than active=False with simply a context `active_test=False` like we have here in 13.0, and also a lot of extra empty lines have been added with no value.

**Comment by etobella** [comment]
@pedrobaeza I think your comments have been attended correctly (at least is what I hope :smile: )

**Comment by pedrobaeza** [comment]
Well, I see very heavy changes here, and knowing that this has been merged in 14.0 doesn't comfort me, as it's removing some features as I can see reading the diff. I also see that the first commit added a demo file that is removed on the second commit, so they should be both merged and neutralized.

**Comment by etobella** [comment]
Well, removed features were unused features. They couldn't be accessed from the agreement and had no sense if they couldn't be edited. However, I think we should need to create some extra features like how to manage renewals and so on. But this should go on a new PR, this intends to have both branches similar

**Comment by pedrobaeza** [comment]
I hope that they are unused features. Please do at least the squash for avoiding useless diffs.

File: agreement_legal/data/agreement_stage.xml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-<?xml version="1.0" ?>
```
Comment (olgamarcocb) [inline agreement_legal/data/agreement_stage.xml:1]: Why are you removing this? You deleted a lot of headers

**Review by olgamarcocb — APPROVED** [review]
:+1:

File: agreement_legal/data/agreement_stage.xml (hunk 1 above)
Comment (pedrobaeza) [reply agreement_legal/data/agreement_stage.xml:1]: Yes, and creating incorrect XML files. They also don't include the encoding, which is necessary.

Comment (etobella) [reply agreement_legal/data/agreement_stage.xml:1]: Fixed, now the headers are correctly formed

**Comment by pedrobaeza** [comment]
Let's not block this anymore, although there were no need to put `utf-8` in upper:

/ocabot merge major

The problem now is to align again 14.0 version reverting the is_old_version thing.

**Pull request merged**